src/components/StepBar/StepBar.test.tsx

```diff
@@ -1,58 +1,123 @@
-  const step2 = getByText(mockSteps[1].title).parentNode;
-  expect(step1).toBeInTheDocument();
-  expect(step2).toBeInTheDocument();
 
-  fireEvent.click(step1);
+  describe('class names', () => {
+    test('should apply StepBar class to the container element', () => {
+      const { container } = render(<StepBar {...DEFAULT_PROPS} />);
+      const stepBar = container.getElementsByClassName(styles.StepBar);
 
-  expect(mockOnChangeStep).toHaveBeenCalledWith(mockSteps[0]);
-});
+      expect(stepBar.length).toBe(1);
+    });
+
+    test('should apply Step class to step elements', () => {
+      const { container } = render(<StepBar {...DEFAULT_PROPS} currentStepId={MOCK_STEPS[1].id} />);
+      const step = container.getElementsByClassName(styles.Step);
+
+      expect(step.length).toBe(1);
+    });
+
+    test('should apply Bar class to only the current step element', () => {
+      const { container } = render(<StepBar {...DEFAULT_PROPS} />);
+      const currentStep = container.getElementsByClassName(styles.Bar);
+
+      expect(currentStep.length).toBe(1);
+    });
+
+    test('should apply className prop to container class', () => {
+      const { container } = render(<StepBar {...DEFAULT_PROPS} className={'mock-class-name'} />);
+      expect(container.firstElementChild).toHaveClass('mock-class-name');
+
+      const stepBar = container.getElementsByClassName('mock-class-name');
+      expect(stepBar.length).toBe(1);
+    });
+  });
 
-test('should render 3rd step', () => {
-  const { getByText } = renderComponent({
-    currentStepId: mockSteps[2].id,
-    steps: mockSteps,
-    onChangeStep: mockOnChangeStep
+  describe('step', () => {
+    test('should apply the step title', () => {
+      render(<StepBar {...DEFAULT_PROPS} currentStepId={MOCK_STEPS[1].id} />);
+
+      expect(screen.getByText(MOCK_STEPS[0].title)).toBeInTheDocument();
+    });
+
+    test('should contain ArrowLink', () => {
+      const { container } = render(<StepBar {...DEFAULT_PROPS} />);
+      const step = container.getElementsByClassName(styles.Step);
+
+      expect(step[0].firstChild).toHaveClass('ArrowLink');
+    });
+
+    test('should call onChangeStep when defined and clicked', () => {
+      render(<StepBar {...DEFAULT_PROPS} currentStepId={MOCK_STEPS[2].id} onChangeStep={mockOnChangeStep} />);
+
+      fireEvent.click(screen.getByText(MOCK_STEPS[1].title));
+      expect(mockOnChangeStep).toHaveBeenCalledWith(MOCK_STEPS[1]);
+
+      fireEvent.click(screen.getByText(MOCK_STEPS[0].title));
+      expect(mockOnChangeStep).toHaveBeenCalledWith(MOCK_STEPS[0]);
+
+      expect(mockOnChangeStep).toHaveBeenCalledTimes(2);
+    });
+
+    test('should not call onChangeStep when undefined and clicked', () => {
+      render(<StepBar {...DEFAULT_PROPS} onChangeStep={() => undefined} />);
+
+      fireEvent.click(screen.getByText(MOCK_STEPS[1].title));
+      expect(mockOnChangeStep).not.toHaveBeenCalled();
+    });
```

Comment:
I think we can omit this one - the component will never call a function it isn't passed.

Reply:
Makes sense!